src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -674,7 +687,10 @@ static int am_recv_event(struct fi_cq_tagged_entry *wc, MPIR_Request * rreq)
     if ((uo_msg = MPIDI_OFI_am_claim_unordered_msg(fi_src_addr, next_seqno)) != NULL) {
         am_hdr = &uo_msg->am_hdr;
         orig_buf = am_hdr;
-        goto repeat;
+#if NEEDS_STRICT_ALIGNMENT
+        need_alignment_check = 0;
```

Comment:
I don't think this correct. When you have a *new* message from the unordered message queue, that message need alignment check, right?

Reply:
when it is an unordered message, it has been aligned in `MPIDI_OFI_am_enqueue_unordered_msg`

Reply:
Oh, I see. I totally misunderstood the code then. And `need_alignment_check` is a wrong name (sorry for suggesting that, but that is the fault of confusing code). The correct name should be something like `has_alignment_copy`. It is not saying that we "need" do alignment check, it is saying that we did check already and we "made" the alignment copy and the buffer is not original, right?

Reply:
right